int64 num
float64[]  aveD
float64[]  minD
float64[]  angle
float64[]  size
